src/js/components/DataTable/DataTable.js

```diff
@@ -67,6 +67,7 @@ const DataTable = ({
   size,
   sort: sortProp,
   sortable,
+  childComponent,
```

Comment:
I think we should just use `children`, rather than adding a separate property for this. This will be more consistent with how InfiniteScroll, Select, Distribution, etc. work.

Also, need to update `doc.js` to explain how `children` will be called.

Reply:
Since the expanded data should be shown for all the rows if required, I feel it'll be better to use it as a property rather than having it as a child as it may confuse users. If it's pagination or InfiniteScroll then I agree that it can be a children instead of property. Let me know your thoughts.

Reply:
Hmm... perhaps you have a point, in that for Distribution, Select, and InfiniteScroll it applies to all items. Whereas, here it applies to only the expanded item. In which case, I think we should relabel the property to something like `rowDetails` rather than `childComponent`.

Reply:
Addressed the change